bugsnag/bugsnag-js
# [PLAT-4935] Add automate-symbolication command (iOS)

Add a command to automate source map upload when a React Native build happens.

Unit tests included for relevant parts. Interactive CLI aspects will be test via maze runner at a later point.

To test this out, use the following steps:

- `git fetch && git checkout feature/rn-cli-automate-symbolication-cmd-ios`
- `npm i`
- `npm run bootstrap`
- `npm run build`
- `npx lerna publish v99.99.99-canary.`git rev-parse HEAD` --no-push --exact --no-git-tag-version --registry http://localhost:4873/` (ensure verdaccio is running, per the react-native/CONTRIBUTING.md guide)
- `cd packages/react-native-cli`
- `npm link` (this symlinks the local cli package to your global npm modules)

On a React Native project:

- check that the command is linked up correctly `bugsnag-react-native-cli --help` 
- install the notifier you published to your local npm `npm i @bugsnag/react-native@v99.99.99-canary.{gitsha} --registry http://localhost:4873/` or `yarn add @bugsnag/react-native@v99.99.99-canary.{gitsha} --registry http://localhost:4873/`
- run `bugsnag-react-native-cli automate-symbolication`
